### DISCUSSION TITLE
How to enable metrics on the AutoscalingRunnerSet or AutoscalingListener?

### DISCUSSION THREAD

**Discussion by alexandrst88** [actions/actions-runner-controller#2894] [FAQs] [answered]
Hi! I'm using, gha-runner-scale-set-controller 0.5.0. The installation method is helm chart from this repository.

I've enabled metrics for the gha-runner-scale-set-controller uncomment this block https://github.com/actions/actions-runner-controller/blob/master/charts/gha-runner-scale-set-controller/values.yaml#L83.

But i was searching with no luck to find how to enable metrics or expose this for listener AutoscalingListener or AutoscalingRunnerSet, to expose other gha_* metrics which are present on this Grafana dashboard https://github.com/actions/actions-runner-controller/tree/master/docs/gha-runner-scale-set-controller/samples/grafana-dashboard.

**Comment by Link-** [comment]
How are you scraping the metrics?

**Reply by alexandrst88** [reply]
From prometheus. but the issue that i can't enable metrics port on the AutoscalingListener or AutoscalingRunnerSet via helm charts or plain crd

**Reply by Link-** [reply]
You need to scrape both the listener (or as many listeners as you have) and the controller manager! There are 2 endpoints

**Reply by alexandrst88** [reply]
the listener is not exposing any ports, but from 0.6.0 seems like is doable to enable port for metrics.

**Reply by Link-** [reply]
@alexandrst88 how come? There must be something wrong with the setup or a bug. The listener and controller manager pods must expose ports and metrics endpoints if configured properly. Share your values.yaml file here after scrubbing the sensitive information.

**Reply by alexandrst88** [reply]
seems like now, it's working. thanks.

**Comment by Link-** [comment, marked as answer]
Documentation for metrics is available in the link below. The listener (or as many as you have configured) and the controller manager pods should be exposing separate metrics ports and endpoints. Make sure to scrape all of them.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/deploying-runner-scale-sets-with-actions-runner-controller#enabling-metrics

**Comment by Charlotte-br560** [comment]
To enable metrics for AutoscalingListener or AutoscalingRunnerSet, you can simply follow a similar approach to what you did for gha-runner-scale-set-controller. Just uncomment the relevant blocks in the values.yaml file, like you did before. If you need further assistance or specific guidance, feel free to check out Crawlbase for comprehensive documentation and support
